src/admin/admin.controller.ts

```diff
@@ -1,4 +1,112 @@
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Get, Param, ParseIntPipe, Post, UnauthorizedException } from '@nestjs/common';
+import { AdminService } from './admin.service';
+import { TransactionService } from 'src/transactions/transactions.service';
+import { UserService } from 'src/user/user.service';
+import { TransactionType, WalletType } from '@prisma/client';
+import { ProviderService } from 'src/provider/provider.service';
 
 @Controller('admin')
-export class AdminController {}
+export class AdminController {
```

Comment:
Good to have a clear JSON response schema defined for every API reponse with status code, message and the actual results of the API call.